ui/src/containers/Search/index.tsx

```diff
@@ -1,19 +1,38 @@
 import { TextInput } from '@patternfly/react-core';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import { useMst } from '../../store/root';
+import { useDebounce } from '../../utils/useDebounce';
 import './Search.css';
 
 const Search: React.FC = () => {
   const { resources } = useMst();
-
   const [value, setValue] = useState('');
 
-  const onSearchChange = (text: string) => {
-    setValue(text);
-    resources.setSearch(text.trim());
+  // to get query param from the url
+  const searchParams = new URLSearchParams(window.location.search);
+  const query = searchParams.get('search_query') || ' ';
+
+  useEffect(() => {
+    if (query !== ' ') {
+      resources.setSearch(query);
+      setValue(query);
+    }
+  }, [query, resources]);
+
+  const setParams = ({ query = '' }) => {
```

Comment:
Not sure if this function should be here or in utils as well and also I think we need to add unit test for this function

Reply:
@PuneetPunamiya I guess this is not a generic function which we can keep in utils and also for testing we'll have to export this function. I guess instead of this if it's possible can we test it's parent caller function?

Reply:
@vinamra28  agreed on not keeping it in utils, but we can keep it in `common` dir as I think params will be required for all pages for e.g login and exporting a function won't be an issue

Reply:
@vinamra28 @PuneetPunamiya while adding params to url for filters I will move it with the filters params  for now we can keep here , WDYT?